### PR TITLE
Update github action to include requirements

### PR DESCRIPTION
**Pull request by karkipy**
- [x] remove need for checkout 

- [x] publish with next tag on push and publish with latest tag on release 

- [x] give grant to install bhoos packages and other npm packages as well

- [x] add readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,21 +3,39 @@
...
+            fetch-depth: 0
+        - name: Use Node.js ${{ matrix.node-version }}
+          uses: actions/setup-node@v1
+          with:
+            node-version: '10.x'
+        - name: Use Bhoos release action
+          uses: bhoos/action-release@v{currentVersion}
+          env:
+            # make sure github.token instead of personal access token is passed else there will be an infinite loop
+            GITHUB_PERSONAL_ACCESS_TOKEN: ${{ github.token }}
```
Comment (syaau) [inline README.md]: @karkipy there is no point in providing github.token to our action which can get this from github context from within

File: index.js [hunk 2]
```diff
@@ -58,10 +77,10 @@ try {
     execSync(`npm version ${branch} -m "Release ${branch} version %s"`);
 
 
-  // build the package and test it
-    execSync(`yarn`);
-    execSync(`yarn build`);
-    execSync(`yarn test`);
+  // test and build
+    execSync(`npm install`);
```
Comment (bhoos-ops) [inline index.js]: @karkipy Why change from yarn? We will be using yarn on all our projects and will depend on `yarn.lock` file.

Comment (karkipy) [reply index.js]: wanted to test a hunch, will revert it

File: README.md (hunk 1 above)
Comment (karkipy) [reply README.md]: yeah that has been fixed

**Pull request merged**